core/src/spotify_id.rs

```diff
@@ -38,16 +39,16 @@ impl SpotifyId {
             n = n + u128::from(d);
         }
 
-        SpotifyId(n)
+        Ok(SpotifyId(n))
     }
 
-    pub fn from_raw(data: &[u8]) -> SpotifyId {
+    pub fn from_raw(data: &[u8]) -> Result<SpotifyId> {
```

Comment:
Thinking about it I'm less bothered about `from_raw` directly returning a `SpotifyId`. It can only go wrong if data isn't 16 bytes, unlike `from_base16`/`from_base62` which can go wrong with invalid characters, also it's an "internal" method.

Reply:
given that everything else now returns a Result, shouldn't this return a result for the sake of consistency?